merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/http/MerlinBindings.java

```diff
@@ -231,6 +236,28 @@ private void validateModelArguments(final Context ctx) {
     }
   }
 
+  private void validatePlan(final Context ctx) {
+    try {
+      final var planId = parseJson(ctx.body(), hasuraPlanActionP).input().planId();
+
+      final var plan = this.planService.getPlan(planId);
+      final var activities = plan.activityInstances.entrySet().stream().collect(Collectors.toMap(
+          Map.Entry::getKey,
+          e -> new SerializedActivity(e.getValue().type, e.getValue().arguments)));
+      final var failures = this.missionModelService.validateActivityInstantiations(plan.missionModelId, activities);
+
+      ctx.result(ResponseSerializers.serializeUnconstructableActivityFailures(failures).toString());
+    } catch (final InvalidJsonException ex) {
+      ctx.status(400).result(ResponseSerializers.serializeInvalidJsonException(ex).toString());
+    } catch (final InvalidEntityException ex) {
+      ctx.status(400).result(ResponseSerializers.serializeInvalidEntityException(ex).toString());
+    } catch (final NoSuchPlanException ex) {
+      ctx.status(404).result(ResponseSerializers.serializeNoSuchPlanException(ex).toString());
+    } catch (final MissionModelService.NoSuchMissionModelException ex) {
+      ctx.status(404);
```

Comment:
Could we add `ctx.status(404).result(ResponseSerializers.serializeNoSuchMissionModelException(ex).toString());` here?

Reply:
Just pushed this change